### DISCUSSION TITLE
[HELP] Filter out children of a Web Component before rendering.

### DISCUSSION THREAD

**Discussion by LePichu** [chialab/dna#102] [Q&A] [answered]
Let's say I have a Web Component which I use in HTML like this:
```html
<material3-button>
    <h1>Test</h1>
</material3-button>
```
However, the final result which is rendered is this:
```html
<material3-button>
    <button></button>
</material3-button>

```

Is there a way to get the children of the web component before it resets them and append said children to the inner content of the web component?

**Comment by LePichu** [comment]
@edoardocavazza should have mentioned you here, you probably didn't see this.

**Comment by edoardocavazza** [comment, marked as answer]
What you are asking is called "slotting". You can use the `<slot />` element in the template to render children:

```tsx
render() {
  return <button>
    <slot />
  </button>;
}
```

You can read more [here](https://www.chialab.io/projects/dna?page=templates#slotted-children).

You can name slots and, maybe, you may be interested to the "[extending native elements](https://www.chialab.io/projects/dna?page=get-started#extending-native-elements)" section. I strongly advice you to extend the button element instead of wrapping it.

**Reply by LePichu** [reply]
I wasn't trying to get button element into the children implicitly, but rather other elements too. Slots are what I was looking for, thanks!

**Reply by LePichu** [reply]
@edoardocavazza Any way to filter certain kinds of elements using Slot? What if I want only divs which have the same slot?
